### ISSUE TITLE
Networking RPC [Remote]

### ISSUE THREAD

**Issue by kwrooijen** [arcadia-unity/ArcadiaGodot#28]
Currently there's no way to do any sort of networking with pure Clojure. We have to define functions in C#.
```csharp
    [Remote]
    public void some_function(string a){
        // Do something
    }
```

It would be nice if we had some sort of interface that we could use to communicate with pure Clojure.

Another problem is that (AFAIK) it doesn't seem like you can send objects through RPC calls (e.g. keywords can't be sent). One way to solve this is to serialize the data we send (EDN) and deserialize it on the other end. Using `prn-str` / `clojure.edn/read-string` will probably be a bit slow. Although I don't know what the alternative would be.

**Comment by kwrooijen** [comment]
I created a separate project to try and solve this problem. The README describes how it can be used: https://github.com/kwrooijen/ArcadiaGodotNetwork

TLDR: This project injects network nodes into your project and  allows to communicate between server and client through multimethods / keyword identifiers. Currently it uses `clojure.edn/read-string` to process the input. Hopefully we can find a better way to solve this.

Let me know what you think. It was too experimental to be added as a PR which is why I created a new project. However I'll be happy to include it in ArcadiaGodot once we have a proper design.
